elcid/assets/js/elcidtest/diagnosis_hospital_number.controller.test.js

```diff
@@ -122,6 +125,18 @@ describe('DiagnosisHospitalNumber', function(){
             $scope.newForPatientWithActiveEpisode(patient);
             expect($scope.addForPatient).toHaveBeenCalled();
         });
+        it('if the patient is marked for follow up we should call the confirm discharge modal', function(){
+            var patient = angular.copy(patientData);
+            patient.episodes['1'].location[0].category = "Followup";
+            $scope.tags = {tag: "infectious_diseases"};
+            spyOn($modal, "open").and.returnValue({result: {then: function(x, y){}}});
+            $scope.newForPatientWithActiveEpisode(patient);
+            expect($modal.open).toHaveBeenCalled();
+            var modalArgs = $modal.open.calls.argsFor(0)[0];
+            expect(modalArgs.templateUrl).toBe('/templates/modals/confirm_discharge.html');
+            expect(modalArgs.controller).toBe('ConfirmDischargeCtrl');
+            expect(modalArgs.resolve.context()).toEqual({ctx: 'some context'});
```

Comment:
I reckon it might be really easy to add checks for the rest of the context here - or in a second test explicitly for the resolves?

patient, episode, tags, nextStepcontroller are all missing tests but should be quite trivial?